tools/toolchains/gcc.py

```diff
@@ -82,6 +86,17 @@ def __init__(self, target,  notify=None, macros=None,
 
         self.flags["common"] += self.cpu
 
+        # Coverage is turned On if coverage_filter is not empty. This means all sources are compiled with coverage
+        # macro and application is linked with coverage flags.
+        # Only source files that match regex from coverage_filter list are compiled with coverage flags. Since turning
+        # On code coverage on all sources increases static data size it overlaps with stack allocation and causes linker
+        # error.
+        self.coverage_filter = coverage_filter
+        if self.coverage_filter:
+            self.flags["common"].append("-g")
+            self.flags["common"].append("-O0")
```

Comment:
@mazimkhan I'm wondering if my suggestion for the compiler flags was such a good idea now, maybe I overlooked something? It may be worth reverting what I suggested.